test/cql-pytest/test_secondary_index.py

```diff
@@ -185,3 +186,19 @@ def test_paging_with_desc_clustering_order(cql, test_keyspace):
             cql.execute(f"INSERT INTO {table}(p,c) VALUES ({i}, 42)")
         stmt = SimpleStatement(f"SELECT * FROM {table} WHERE c = 42", fetch_size=1)
         assert len([row for row in cql.execute(stmt)]) == 3
+
+# Selecting values using only clustering key should require filtering, but work correctly
+# Reproduces issue #8991
+@pytest.mark.xfail(reason="Select from indexed table using only clustering key breaks. Issue #8991")
+def test_filter_cluster_key(cql, test_keyspace):
+    schema = 'p int, c1 int, c2 int, primary key (p, c1, c2)'
+    extra = ''
+    with new_test_table(cql, test_keyspace, schema, extra) as table:
+        cql.execute(f"CREATE INDEX ON {table}(c2)")
+        for c1 in [0, 1, 2]:
+            for c2 in [0, 1, 2]:
+                cql.execute(f"INSERT INTO {table} (p, c1, c2) VALUES (0, {c1}, {c2})")
```

Comment:
Nitpick: do we really need 3*3 rows for this test to fail? You can perhaps shave off a few milliseconds by using fewer rows.

Reply:
To reproduce this specific issue two rows would be enough, my thinking was that if there are more rows we could possibly catch more cases of similar bugs in the future.

I wil reduce this to just these two rows